### PR TITLE
feat!: oracle role CHAIN-444

### PR DESCRIPTION
**Pull request by crystalbit**
## ChangeLog
- Changes around setFLTPrice - it was onlyOwner, now it can be triggered only with oracle
- added oracle role (`setOracle(address) onlyOwner`, `oracle() view`)

### CODE REVIEW

File: src/core/interfaces/ICapacityConst.sol [hunk 1]
```diff
@@ -15,6 +15,10 @@ interface ICapacityConst is IEpochController {
     /// @param newValue The new value of the constant
     event CapacityConstantUpdated(CapacityConstantType constantType, uint256 newValue);
 
+    /// @dev Emitted then new oracle address is set
```
Comment (AlcibiadesCleinias) [inline src/core/interfaces/ICapacityConst.sol:18]: when*

File: src/core/CapacityConst.sol [hunk 2]
```diff
@@ -217,8 +222,9 @@ contract CapacityConst is ICapacityConst, OwnableUpgradableDiamond, EpochControl
     // #endregion ------------------ External View Functions ------------------
 
     // #region ------------------ External Mutable Functions ------------------
-    function setFLTPrice(uint256 fltPrice_) public onlyOwner {
+    function setFLTPrice(uint256 fltPrice_) public {
```
Comment (AlcibiadesCleinias) [inline src/core/CapacityConst.sol:225]: not sure if we want. But from code prospective it is better to be flt agnostic in contract namings. Rly not sure

Comment (AlcibiadesCleinias) [reply src/core/CapacityConst.sol:225]: ok, I reread seems we want to leave as it is

File: src/core/interfaces/ICapacityConst.sol (hunk 1 above)
Comment (crystalbit) [reply src/core/interfaces/ICapacityConst.sol:18]: oh :) ty

File: src/core/CapacityConst.sol (hunk 2 above)
Comment (crystalbit) [reply src/core/CapacityConst.sol:225]: I think so too, FLT is already everywhere and isn't going to be changed

**Review by AlcibiadesCleinias — APPROVED** [review]

**Pull request merged**